### DISCUSSION TITLE
Improvement of regions_to_network performance (already implemented in a fork)

### DISCUSSION THREAD

**Discussion by Arenhart** [PMEAL/porespy#956] [Ideas]
Hi,

We have been working on the regions_to_network function, improving it's performance with Numba. We got an improvement of up to x60 faster network extraction when running in parallel. We also modified it to allow for anisotropic volumes.

I'm committing in our fork, but I would like to know if it would benefit the main repository. If so, what would be the requirements for opening a PR? It is still missing a code cleanup, tests, and more results validation.

While we tried to keep the results as close as possible to the original function, some differences were included, listed as following:
- Only valid for 3D volumes.
- Throat and perimeter calculations modified to allow anisotropic volumes (in a very simplified explanation, marching squares is applied to a flattened throat image).
- Distance local and global maxima is the average of all coordinates that have the highest distance transform value.
- Changed the marching cubes algorithm to an implementation that outputs the area and volume without the need of creating the mesh, "standard" accuracy was removed due to similar calculation time.

**Comment by ma-sadeghi** [comment]
That would be very welcome! As for the requirements, I guess feel free to open the PR and we can iterate on it then.

**Comment by jgostick** [comment]
This seems like a very significant and helpful contribution!  Would you perhaps be open to a teleconference where you can explain your code and logic in a proper high-bandwidth conversation?  

BTW, parallelizing this function has been on our "todo" since it was written, but have never found the time.  So this contribution is really the entire point of "open source", where we can all contribute to something together.  Anyway, very exciting.  Please email me directly at my uwaterloo address if you are willing to have a 30 minute meeting (jgostick@...).
